aspnetcore/fundamentals/middleware/index.md

```diff
@@ -199,6 +190,23 @@ app.Map("/level1", level1App => {
 
 [!code-csharp[](index/snapshot/Chain/StartupMultiSeg.cs?highlight=13)]
 
+<xref:Microsoft.AspNetCore.Builder.MapWhenExtensions.MapWhen*> branches the request pipeline based on the result of the given predicate. Any predicate of type `Func<HttpContext, bool>` can be used to map requests to a new branch of the pipeline. In the following example, a predicate is used to detect the presence of a query string variable `branch`:
+
+[!code-csharp[](index/snapshot/Chain/StartupMapWhen.cs?highlight=14-15)]
+
+The following table shows the requests and responses from `http://localhost:1234` using the previous code:
+
+| Request                       | Response                     |
+| ----------------------------- | ---------------------------- |
+| localhost:1234                | Hello from non-Map delegate. |
+| localhost:1234/?branch=master | Branch used = master         |
+
+<xref:Microsoft.AspNetCore.Builder.UseWhenExtensions.UseWhen*> also branches the request pipeline based on the result of the given predicate. Unlike with `MapWhen`, this branch is rejoined to the main pipeline if it does short-circuit or contain a terminal middleware:
+
+[!code-csharp[](index/snapshot/Chain/StartupUseWhen.cs?highlight=23-24)]
+
+In the preceding example, a response of "Hello from main pipeline." is written for all requests. If the request includes a query string variable `branch`, its value is logged before the main pipeline is rejoined.
```

Comment:
I shortened this up.

Reply:
Yeah, that's definitely better. I overdid it with the short sentence thing. Thanks.